api/control/v1alpha1/types.go

```diff
@@ -260,11 +260,19 @@ type BindingSpec struct {
 type DownsyncObjectReferences struct {
 	// `clusterScope` holds a list of cluster-scoped object references with their associated
 	// resource versions to downsync.
-	ClusterScope []ClusterScopeDownsyncObject `json:"clusterScope,omitempty"`
+	ClusterScope []ClusterScopeDownsyncObjectAndStatusCollectors `json:"clusterScope,omitempty"`
 
 	// `namespaceScope` holds a list of namespace-scoped object references with their associated
 	// resource versions to downsync.
-	NamespaceScope []NamespaceScopeDownsyncObject `json:"namespaceScope,omitempty"`
+	NamespaceScope []NamespaceScopeDownsyncObjectAndStatusCollectors `json:"namespaceScope,omitempty"`
+}
+
+// NamespaceScopeDownsyncObjectAndStatusCollectors represents a specific namespace-scoped object to downsync,
+// and the status collectors that should be applied to it.
+type NamespaceScopeDownsyncObjectAndStatusCollectors struct {
+	*NamespaceScopeDownsyncObject `json:",inline"`
```

Comment:
The level of pointer indirection here is surprising/unusual. In a kube API object, pointer indirection is usually associated with a bit of substructure being optional. But that is not what is happening here.

For an example in Kubernetes, consider the slice of Container in a PodSpec (Container is to PodSpec as NamespaceScopeDownsyncObjectAndStatusCollectors is to DownsyncObjectReferences).  A Container has a non-optional field `Resources ResourceRequirements` --- no pointer.

Reply:
Thanks